compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -606,6 +606,7 @@ object Contexts {
     /** A table for hash consing unique types */
     private[core] val uniques = new util.HashSet[Type](Config.initialUniquesCapacity) {
       override def hash(x: Type): Int = x.hash
+      override def isEqual(x: Type, y: Type) = x.eql(y)
```

Comment:
Shouldn't the override also be used in the other subclasses of `HashSet` ? That is, `AppliedUniques` and `NamedTypeUniques`

Reply:
No, `equals` is never called on those sets. Everything is based on `eq` for them.